airgun/entities/host_new.py

```diff
@@ -32,6 +38,163 @@ def get_details(self, entity_name, widget_names=None):
         self.browser.plugin.ensure_page_safe()
         return view.read(widget_names=widget_names)
 
+    def edit_system_purpose(
+        self, entity_name, role=None, sla=None, usage=None, release_ver=None, add_ons=None
+    ):
+        """
+        Function that edits the system purpose of a host
+
+        Args:
+            entity_name: Name of the host
+            role: Role to be assigned
+            sla: SLA to be assigned
+            usage: Usage to be assigned
+            release_ver: Release version to be assigned
+            add_ons: Add-ons to be assigned
+
+        Raises:
+            ValueError: If no parameters are passed.
+        """
+
+        if not any([role, sla, usage, release_ver, add_ons]):
+            raise ValueError(
+                'At least one of the role, sla, usage, release_ver, add_ons ' 'must be provided!'
+            )
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        view.overview.system_purpose.edit_system_purpose.click()
+        view = EditSystemPurposeView(self.browser)
+        self.browser.plugin.ensure_page_safe()
+        view.wait_displayed()
+        if role:
+            view.role.fill(role)
+        if sla:
+            view.sla.fill(sla)
+        if usage:
+            view.usage.fill(usage)
+        if release_ver:
+            view.release_version.fill(release_ver)
+        if add_ons:
+            view.add_ons.fill(add_ons)
+        view.save.click()
+
+    def add_host_to_host_collection(
+        self, entity_name, host_collection_name=None, add_to_all_collections=False
+    ):
+        """
+        Function that adds host to host collection
+
+        Args:
+            entity_name: Name of the host
+            host_collection_name: Name of the host collection we want to remove host from
+            add_to_all_collections: If True, host will be added to all host collections
+
+        Raises:
+            ValueError: If specific hostColName is set to be removed and add all is set to True.
+            ValueError: If no parameters are passed.
+            ValueError: If host is already assigned to selected host collection.
+            ValueError: If there are no host collections left for addition.
+            ValueError: Given host collection name is not found in host collections.
+        """
+
+        if host_collection_name and add_to_all_collections:
+            raise ValueError(
+                'Either host_collection_name or add_to_all_collections must be provided!'
+            )
+        if (host_collection_name is None) and (add_to_all_collections is False):
+            raise ValueError(
+                'host_collection_name None and add_to_all_collections False '
```

Comment:
This message is too general. Can't we use these two conditions (both specified or neither specified) in one condition and in either case, just show `Either host_collection_name or add_to_all_collections must be provided!`?

Reply:
There is another line in that error saying that it is an invalid combination.

I wanted to do that as one condition but I also didn't want it to look like this:
```python
if (host_collection_name and add_to_all_collections) or ((host_collection_name is None) and (add_to_all_collections is False)):
    pass
```

so I chose this approach.

Reply:
The above looks nice to me, if correctly split in lines and if `pass` is replaced by `raise...`

Reply:
I can split it then.

with `pass` in that comment I was just lazy :D